db/models/aggregate_functions.py

```diff
@@ -153,9 +153,31 @@ def submit_application(application_id):
 def update_form(application_id, form_name, question_json):
     try:
         form_sql_row = FormsMethods.get_form(application_id, form_name)
+        # Running update form for the first time
+        if question_json and not form_sql_row.json:
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+        # Removing all data in the form
+        elif form_sql_row.json and not question_json:
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+        # Updating form subsequent times
+        elif (
+            form_sql_row.json
+            and form_sql_row.json[0]["fields"] != question_json[0]["fields"]
+        ):
+            ApplicationsMethods.application_edited(application_id)
+            current_app.logger.info(
+                f"Application updated for application_id: '{application_id}."
+            )
+    except sqlalchemy.orm.exc.NoResultFound as e:
```

Comment:
is it possible to use pattern matching?

Reply:
Okay if its too much hassle then probably not worth it but if it is doable then give it go: 😆